lib/client.js

```diff
@@ -1938,6 +1938,37 @@ MatrixClient.prototype.resolveRoomAlias = function(roomAlias, callback) {
 
 /**
  * @param {string} roomId
+ * @param {module:client.callback} callback Optional.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.getRoomVisibility = function(roomId, callback) {
```

Comment:
The name of this isn't immediately clear this refers to the room directory listing. Whilst I appreciate the parity with the `/createRoom` API which sets this, the `visibility` key there does a lot more than *just* that (`join_rules` as well!). Given that, I would prefer it if you could thwack a "directory" in there somewhere to make it clear it is for the directory rather than say, a shortcut for `join_rules`. `getRoomDirectoryVisibility` perhaps?